### ISSUE TITLE
Discord - Roles Not Being Fully Applied 

### ISSUE THREAD

**Issue by kanelarrete** [warlof/seat-connector#56]
Running into a weird situation where roles are no longer being applied to a point. It will work down the list hitting some but not others. Checking Horizon logs, I am getting the following error.

```
Symfony\Component\Debug\Exception\FatalThrowableError: Return value of Warlof\Seat\Connector\Models\User::buildConnectorNickname() must be of the type string, null returned in /var/www/seat/vendor/warlof/seat-connector/src/Models/User.php:245
--
```

Steps I have done at this point:

I have checked the seat_connector_users table for any null names and there were none. I have checked users table for null names and none there as well. I have tried apply user level roles to users above and below them (via their ID # assuming it does the sync sequentially) and while it would apply to the one above it would not the one below _but_ would I could with others down the line. 

At a bit of a road block here so any help would be appreciated. Thanks!

**Comment by warlof** [comment]
Does all your mapped user have at least one character linked ?

What's your ticker setup?

**Comment by kanelarrete** [comment]
From what I can tell from both the frontend and through SQL they all have at least one character linked.

Use Ticker is selected to no but the mask is [%2$s] %1$s

**Comment by kanelarrete** [comment]
Figured it out. You were correct on your hunch about the mapped users having at least one character linked. I did end up finding a few where the UI was showing that they indeed have a character but if you looked that character up in the users section of seat, they were not clickable (deleted?). Removing those connections allowed the process to run again. In SeAT v3, there was the ability to see invalid tokens via the users section but looks like it has since been removed in SeAT v4 which could have potentially been helpful in this case. 

Thanks again and appreciate you pointing me in the correct direction.
